controller/machine/controller.go

```diff
@@ -53,6 +54,7 @@ type Lifecycle struct {
 	machineTemplateClient        v3.MachineTemplateInterface
 	configMapGetter              typedv1.ConfigMapsGetter
 	logger                       event.Logger
+	clusters                     v3.ClusterInterface
```

Comment:
we should use cache here (the lister). Unless there is a need to pull from api directly